Limit sanity checks by Python 3.6+

tkrishtop opened the pull request:
Limit ansible-test checks to Python versions from 3.6 to 3.9 available from the ansible image.

tkrishtop commented:
We could pin the Python version using --python argument:
```
ansible-test sanity --verbose --docker --python 3.6 --color --coverage --failure-ok
```

I'm currently trying to go through several Python versions by using a loop, something like 
```
for version in 3.6 3.7 3.8 3.9 ; do
    ansible-test sanity --verbose --docker --python $version --color --coverage --failure-ok
done
```

tkrishtop commented:
I managed to limit `ansible-test` checks by a list of pre-defined Python versions. I used 3.6+ versions available in the docker image: from 3.6 to 3.9 included.

The checks `stable-2.10` and `stable-2.15` are now failing because of the comparison logic between `branch` and `main`. It should be trimmed for the output of a `for` loop I introduced.

tkrishtop commented:
The checks stable-2.10 and stable-2.15 are now green but I wonder how to fix ansible-lint score. Do I need to add a changelog even for the .gihthub/workflow changes? @tonyskapunk could you please have a look?

fredericlepied commented:
> The checks stable-2.10 and stable-2.15 are now green but I wonder how to fix ansible-lint score. Do I need to add a changelog even for the .gihthub/workflow changes? @tonyskapunk could you please have a look?

you can try to add `jmespath` line 27.

tonyskapunk commented on .github/workflows/pr.yml:
```diff
@@ -81,10 +81,13 @@ jobs:
-          $cmd 2>&1 | tee branch.output
+          for version in 3.6 3.7 3.8 3.9 ; do
+            ansible-test sanity --verbose --docker --python $version --color --coverage --failure-ok
+          done 2>&1 | tee branch.output
```
Only 3.9 is being kept in the output, you want to append instead
```suggestion
          done 2>&1 | tee -a branch.output
```

tonyskapunk commented on .github/workflows/pr.yml:
```diff
@@ -81,10 +81,13 @@ jobs:
-          $cmd > main.output 2>&1
+          for version in 3.6 3.7 3.8 3.9 ; do
+            ansible-test sanity --verbose --docker --python $version --color --coverage --failure-ok
+          done > main.output 2>&1
```
likewise, rather append:
```suggestion
          done >> main.output 2>&1
```

fredericlepied replied:
started a fix for that at https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/102

tkrishtop commented:
> > The checks stable-2.10 and stable-2.15 are now green but I wonder how to fix ansible-lint score. Do I need to add a changelog even for the .gihthub/workflow changes? @tonyskapunk could you please have a look?
> 
> you can try to add `jmespath` line 27.

thanks @fredericlepied it made things better.
The current error is because the filepath is not considered in the diff and the same errors but from different files are somehow messed up. I'm checking how to fix the regex.

```
## Improvements over main branch:
+FIXED: roles/vendors/kvm/tasks/exists.yml:1: warning[outdated-tag]: Replaced outdated tag 'unnamed-task' with 'name', replace it to avoid future errors
+FIXED: roles/vendors/supermicro/tasks/iso.yml:31: warning[outdated-tag]: Replaced outdated tag 'var-spacing' with 'jinja', replace it to avoid future errors
## Regressions from main branch:
-ERROR: roles/vendors/hpe/tasks/iso.yml:10: warning[outdated-tag]: Replaced outdated tag 'unnamed-task' with 'name', replace it to avoid future errors
-ERROR: roles/vendors/supermicro/tasks/disk.yml:56: warning[outdated-tag]: Replaced outdated tag 'var-spacing' with 'jinja', replace it to avoid future errors
> Fix regressions listed above
```

tkrishtop commented on .github/workflows/pr.yml:
```diff
@@ -24,12 +24,14 @@ jobs:
         run: |
           set -x
+          # fix randomness
+          export PYTHONHASHSEED=42
```
wow, what a fix! and it works!

tonyskapunk reviewed (commented):
Oops it merged too quick, it needs some rework otherwise we'll hide some results
